packages/earljs/test/matchers/ArrayWith.test.ts

```diff
@@ -60,12 +59,7 @@ describe('ArrayWith matcher', () => {
         earlExpect({ arr: [1, 2, 3] }).toEqual({
           arr: earlExpect.arrayWith(earlExpect.numberCloseTo(6, { delta: 1 })),
         }),
-      ).to.throw(
-        [
-          '{"arr": [1, 2, 3]} not equal to {"arr": "[ArrayWith: [NumberCloseTo: 6, delta=1]]"}',
-          'Hint: value mismatch',
-        ].join(EOL),
-      )
+      ).to.throw('{ arr: [1, 2, 3] } not equal to { arr: Matcher }')
```

Comment:
I think that the previous error msg was actually more helpful....

Reply:
This is because the check here ignores the string diff:
![image](https://user-images.githubusercontent.com/17070569/149503558-0f4a51ca-7278-4da2-9289-05401af7de13.png)

The way matchers are stringified could definitely use some improvement, but I thought the PR is large enough. If this isn't a deal breaker it would be great to get this merged :)